libass/ass_parse.c

```diff
@@ -632,7 +644,7 @@ char *parse_tag(ASS_Renderer *render_priv, char *p, char *end, double pwr)
         if (t1 == -1 && t4 == -1) {
             t1 = 0;
             t4 = render_priv->state.event->Duration;
-            t3 = t4 - t3;
+            t3 = t4 - FFMAX(t3, 0);  // avoid overflow
```

Comment:
Hmm, I think I see what you mean. I guess this needs a bigger comment that explains what’s going on?